### PR TITLE
8261366: Add discussion of IEEE 754 to BigDecimal

### PR DESCRIPTION
**Pull request by jddarcy**
Informative update to BigDecimal and related classes to latest IEEE 754 terminology, including a discussion of the similarities and differences of BigDecimal and IEEE 754 decimal arithmetic.

Once the wording is finalized, I'll reflow any modified paragraphs and update the copyright years, etc.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8261366](https://bugs.openjdk.java.net/browse/JDK-8261366): Add discussion of IEEE 754 to BigDecimal

### Reviewers
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**) ⚠️ Review applies to a29d60fdd6a99ab1567c3f0621c7fa572388f37a

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2868/head:pull/2868`
`$ git checkout pull/2868`

### CODE REVIEW

File: src/java.base/share/classes/java/math/BigDecimal.java [hunk 1]
```diff
@@ -220,6 +220,63 @@
...
+ *
+ * <p>For differences, IEEE 754 includes several kinds of values not
+ * modeled by {@code BigDecimal} including negative zero, signed
+ * infinities, and NaN (not-a-number). IEEE 754 defines formats, which
+ * are parameterized by base (binary or decimal), number of digits of
+ * precision, and exponent range. A format determines the set of
+ * representable values. Most operations accept as input one or more
+ * values of a given format and produce a result in the same format.
+ * A {@code BigDecimal}'s {@linkplain scale() scale} is equivalent to
+ * negating an IEEE 754 value's exponent. {@code BigDecimal} values to
```
Comment (efge) [inline src/java.base/share/classes/java/math/BigDecimal.java:250]: _to_ -> _do_

File: src/java.base/share/classes/java/math/BigDecimal.java [hunk 2]
```diff
@@ -220,6 +220,63 @@
...
+ * not have a format in the same sense; all values have the same
+ * possible range of scale/exponent and the {@linkplain
+ * unscaledValue() unscaled value} has arbitrary precision. Instead,
+ * for the {@code BigDecimal} operations taking a {@code MathContext}
+ * parameter, if the {@code MathContext} has a nonzero precision, the
+ * set of possible representable values for the result is determined
+ * by the precision of the {@code MathContext} argument. For example
+ * in {@code BigDecimal}, if a nonzero three-digit number and a
+ * nonzero four-digit number are multiplied together in the context of
+ * a {@code MathContext} object has a precision of three, the result
```
Comment (efge) [inline src/java.base/share/classes/java/math/BigDecimal.java:260]: _has_ -> _having_

File: src/java.base/share/classes/java/math/BigDecimal.java [hunk 3]
```diff
@@ -220,6 +220,63 @@
...
+ * attributes.
+
+ * <p>{@code BigDecimal} arithmetic will most resemble IEEE 754
+ * decimal arithmetic if a {@code MathContext} corresponding to an
+ * IEEE 754 decimal format, such as {@linkplain MathContext#DECIMAL64
+ * decimal64} or {@linkplain MathContext#DECIMAL128 decimal128} is
+ * used to round all starting values and intermediate operations. The
+ * numerical values computed can differ if the exponent range of the
+ * IEEE 754 format being approximated is exceeded since a {@code
+ * MathContext} does not curtail the scale {@code BigDecimal}
```
Comment (efge) [inline src/java.base/share/classes/java/math/BigDecimal.java:275]: _the scale_ -> _the scale of_

Suggest using another word than _curtail_, whose meaning will not be immediately clear to non-native readers as it's not a technical term.

File: src/java.base/share/classes/java/math/BigDecimal.java [hunk 4]
```diff
@@ -220,6 +220,63 @@
...
+ * results. Operations that would generate a NaN or exact infinity,
+ * such as dividing by zero, throw an {@code Arithmetic-Exception} in
```
Comment (efge) [inline src/java.base/share/classes/java/math/BigDecimal.java:277]: `Arithmetic-Exception` -> `ArithmeticException`

**Review by bplb — APPROVED** [review]
Looks fine.

**Pull request closed without merging**